### PR TITLE
fix(gatsby-source-drupal): Image cdn fixes

### PR DESCRIPTION
**Pull request by TylerBarnes**
This PR does 2 things:
1. Warns about corrupt files when using Image CDN and skips over them.
2. Only registers the gatsbyImage field when skipFileDownloads is enabled. If it's not enabled it prints out a message with info about how to enable Gatsby Image CDN for Drupal.

Relates to https://github.com/gatsbyjs/gatsby/discussions/35147#discussioncomment-2690358

### CODE REVIEW

**Comment by j0sh77** [comment]
Looks good to me, pending the passing of non-flaky tests. 😄

File: packages/gatsby-source-drupal/src/normalize.js [hunk 1]
```diff
@@ -100,6 +100,14 @@ const getGatsbyImageCdnFields = async ({
 
     return gatsbyImageCdnFields
   } catch (e) {
+    if (e.message.includes(`unrecognized file format`)) {
+      reporter.error(
+        `Encountered corrupt file while requesting image dimensions for ${url}`
```
Comment (KyleAMathews) [inline packages/gatsby-source-drupal/src/normalize.js:105]: Add a prefix saying this is from source-drupal as well?

Comment (TylerBarnes) [reply packages/gatsby-source-drupal/src/normalize.js:105]: Good call!

**Review by KyleAMathews — APPROVED** [review]
🚢 !

**Pull request merged**